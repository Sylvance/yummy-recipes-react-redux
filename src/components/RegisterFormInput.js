import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class RegisterFormInput extends Component {
  static propTypes = {
    onSave: PropTypes.func.isRequired,
    username: PropTypes.string,
    email: PropTypes.string,
    password: PropTypes.string,
    firstname: PropTypes.string,
    lastname: PropTypes.string
  }

  state = {
    username: this.props.username || '',
    email: this.props.email || '',
    password: this.props.password || '',
    firstname: this.props.firstname || '',
    lastname: this.props.lastname || ''
  }

  handleSubmit = e => {
    e.preventDefault();

    const { username, email, password, firstname, lastname } = this.state;
    this.props.onSave(username, email, password, firstname, lastname)
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  render() {
    return (
        <form name="form" onSubmit={this.handleSubmit}>
            <input type="text"
                placeholder="Enter your username"
                autoFocus="true"
                name="username"
                value={this.state.username}
                onChange={this.handleChange}
                required />
            <br />
            <input type="email"
                placeholder="Enter your email"
                autoFocus="true"
                name="email"
                value={this.state.email}
                onChange={this.handleChange}
                required />
            <br />
            <input type="password"
                placeholder="Enter your password"
                autoFocus="true"
                name="password"
                value={this.state.password}
                onChange={this.handleChange}
                required />
            <br />
            <input type="text"
                placeholder="Enter your firstname"
                autoFocus="true"
                name="firstname"
                value={this.state.firstname}
                onChange={this.handleChange}
                required />
            <br />
            <input type="text"
                placeholder="Enter your lastname"
                autoFocus="true"
                name="lastname"
                value={this.state.lastname}
                onChange={this.handleChange}
                required />
            <br />
            <input type="submit" value="Register" />
        </form>
    )
  }
}
