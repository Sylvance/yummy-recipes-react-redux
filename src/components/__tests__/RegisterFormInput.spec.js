// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import RegisterFormInput from '../RegisterFormInput';

describe('RegisterFormInput test', () => {
  it('RegisterFormInput should match snapshot', () => {
    const component = renderer.create(
      <RegisterFormInput
        onSave={() => {}}
        username={'defaultString'}
        email={'defaultString'}
        password={'defaultString'}
        firstname={'defaultString'}
        lastname={'defaultString'}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
