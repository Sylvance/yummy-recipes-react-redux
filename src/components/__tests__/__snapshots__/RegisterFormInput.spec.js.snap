// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterFormInput test RegisterFormInput should match snapshot 1`] = `
<form
  className=""
  name="form"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="username"
    >
      Username
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="username"
      onChange={[Function]}
      placeholder="Enter your username"
      required={true}
      type="text"
      value="defaultString"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="email"
    >
      Email
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="email"
      onChange={[Function]}
      placeholder="Enter your email"
      required={true}
      type="email"
      value="defaultString"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="password"
      onChange={[Function]}
      placeholder="Enter your password"
      required={true}
      type="password"
      value="defaultString"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="firstname"
    >
      First name
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="firstname"
      onChange={[Function]}
      placeholder="Enter your first name"
      required={true}
      type="text"
      value="defaultString"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="lastname"
    >
      Last Name
    </label>
    <input
      autoFocus="true"
      className="form-control"
      name="lastname"
      onChange={[Function]}
      placeholder="Enter your last name"
      required={true}
      type="text"
      value="defaultString"
    />
  </div>
  <button
    className="btn btn-success"
    onClick={[Function]}
    type={undefined}
  >
    Register
  </button>
</form>
`;
